plugins/module_utils/linode_common_info.py

```diff
@@ -41,6 +42,28 @@ class InfoModuleParam:
     type: FieldType
 
 
+class InfoModuleParamGroupPolicy(Enum):
+    """
+    Defines the policies that can be set for a param group.
+    """
+
+    EXACTLY_ONE_OF = "exactly_one_of"
+
+
+class InfoModuleParamGroup:
```

Comment:
This addition allows for required but mutually exclusive params (e.g. `domain` and `domain_id`)